internal/controller/dnsrecord_controller.go

```diff
@@ -302,15 +324,27 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 
 	plan = plan.Calculate()
 
+	dnsRecord.Status.ValidFor = defaultRequeueTime.String()
+	dnsRecord.Status.QueuedAt = reconcileStart
 	if plan.Changes.HasChanges() {
+		// generation has not changed but there are changes.
+		// implies that they were overridden - bump write counter
+		if !generationChanged(dnsRecord) {
+			dnsRecord.Status.WriteCounter++
+			logger.V(3).Info("Changes needed on the same generation of record")
+		}
+		dnsRecord.Status.ValidFor = validationRequeueTime.String()
+		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, "AwaitingValidation", "Awaiting validation")
 		logger.Info("Applying changes")
 		err = registry.ApplyChanges(ctx, plan.Changes)
 		if err != nil {
-			return err
+			return validationRequeueTime, err
 		}
 	} else {
 		logger.Info("All records are already up to date")
+		dnsRecord.Status.WriteCounter = 0
+		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionTrue, "ProviderSuccess", "Provider ensured the dns record")
 	}
 
-	return nil
+	return defaultRequeueTime, nil
```

Comment:
nit: To me the logic around updating the conditions and deciding what this requeue time value should be is not the concern of `applyChanges`. This should just return an error and a bool to say if it had changes, and wherever this is being called use that info to calculate the status and requeue validation time.

It also looks like you are setting stuff in the status before you even know if it successfully updated or not.